gen/vsup_web

```diff
@@ -74,59 +74,84 @@ my $groups;  # $groups->{ID}->{NAME/USERS} = name/{uco} = 1;
 
-			my %uAttributes = attributesToHash $member->getAttributes;
+			my $excluded = $data->getMemberAttributeValue( member => $memberId, attrName => $A_EXCLUDED );
 
-			if ($uAttributes{$A_EXCLUDED}) {
+			if ($excluded) {
 				# skip users, which shouldn't be displayed on the web
 				next;
 			}
 
-			my $uco = $uAttributes{$A_UCO};
-			$users->{$uco}->{$A_LOGIN} = $uAttributes{$A_LOGIN};
-			$users->{$uco}->{'EMAIL'} = ($uAttributes{$A_VSUP_PREF_MAIL} || ($uAttributes{$A_VSUP_MAIL} || $uAttributes{$A_LOGIN} . '@vsup.cz'));
+			my $uco  = $data->getUserAttributeValue(member => $memberId, attrName => $A_UCO);
+			my $login  = $data->getUserFacilityAttributeValue( member => $memberId, attrName => $A_LOGIN);
+			my $prefMail = $data->getUserAttributeValue(member => $memberId, attrName => $A_VSUP_PREF_MAIL);
+			my $vsupMail = $data->getUserAttributeValue(member => $memberId, attrName => $A_VSUP_MAIL);
+			my $titleBefore = $data->getUserAttributeValue(member => $memberId, attrName => $A_TITLE_BEFORE);
+			my $artisticFirstName = $data->getUserAttributeValue(member => $memberId, attrName => $A_ARTISTIC_FIRST_NAME);
+			my $firstName = $data->getUserAttributeValue(member => $memberId, attrName => $A_FIRST_NAME);
+			my $artisticLastName = $data->getUserAttributeValue(member => $memberId, attrName => $A_ARTISTIC_LAST_NAME);
+			my $lastName = $data->getUserAttributeValue(member => $memberId, attrName => $A_LAST_NAME);
+			my $titleAfter = $data->getUserAttributeValue(member => $memberId, attrName => $A_TITLE_AFTER);
+			my $studyId = $data->getUserAttributeValue(member => $memberId, attrName => $A_STUDY_ID);
+			my $studySince = $data->getUserAttributeValue(member => $memberId, attrName => $A_STUDY_SINCE);
+			my $studyRocnik = $data->getUserAttributeValue(member => $memberId, attrName => $A_STUDY_ROCNIK);
+			my $studyPrihlaska = $data->getUserAttributeValue(member => $memberId, attrName => $A_STUDY_PRIHLASKA);
+			my $birthDate = $data->getUserAttributeValue(member => $memberId, attrName => $A_BIRTH_DATE);
+			my $foto = $data->getUserAttributeValue(member => $memberId, attrName => $A_FOTO);
+			my $phone = $data->getUserAttributeValue(member => $memberId, attrName => $A_PHONE);
+			my $mobile = $data->getUserAttributeValue(member => $memberId, attrName => $A_MOBILE);
+			my $emailPriv = $data->getUserAttributeValue(member => $memberId, attrName => $A_EMAIL_PRIV);
+			my $emailPrivKOS = $data->getUserAttributeValue(member => $memberId, attrName => $A_EMAIL_PRIV_KOS);
+			my $phonePriv = $data->getUserAttributeValue(member => $memberId, attrName => $A_PHONE_PRIV);
+			my $job = $data->getUserAttributeValue(member => $memberId, attrName => $A_JOB);
+			my $jobId = $data->getUserAttributeValue(member => $memberId, attrName => $A_JOB_ID);
+			my $ns = $data->getUserAttributeValue(member => $memberId, attrName => $A_NS);
+			my $userId = $data->getUserAttributeValue(member => $memberId, attrName => $A_USER_ID);
```

Comment:
We can use ``getUserIdForMember()`` and remove ``user:core:id`` attribute from the required.

Reply:
Fixed